lib/OptimizationEvolutionary/test/runtests.jl

```diff
@@ -56,4 +57,102 @@ Random.seed!(1234)
     # Make sure that both the user's trace record value, as well as `curr_u` are stored in the trace.
     @test haskey(sol.original.trace[end].metadata, "TESTVAL") &&
           haskey(sol.original.trace[end].metadata, "curr_u")
+
+    # Test Suite for Different Multi-Objective Functions
+function test_multi_objective(func, initial_guess)
+    # Define the gradient function using ForwardDiff
+    function gradient_multi_objective(x, p=nothing)
+        ForwardDiff.jacobian(func, x)
+    end
+
+    # Create an instance of MultiObjectiveOptimizationFunction
+    obj_func = MultiObjectiveOptimizationFunction(func, jac=gradient_multi_objective)
+
+    # Set up the evolutionary algorithm (e.g., NSGA2)
+    algorithm = OptimizationEvolutionary.NSGA2()
+
+    # Define the optimization problem
+    problem = OptimizationProblem(obj_func, initial_guess)
+
+    # Solve the optimization problem
+    result = solve(problem, algorithm)
+    
+    return result
+end
+
+@testset "Multi-Objective Optimization Tests" begin
+
+    # Test 1: Sphere and Rastrigin Functions
+    @testset "Sphere and Rastrigin Functions" begin
+        function multi_objective_1(x, p=nothing)::Vector{Float64}
+            f1 = sum(x .^ 2)  # Sphere function
+            f2 = sum(x .^ 2 .- 10 .* cos.(2π .* x) .+ 10)  # Rastrigin function
+            return [f1, f2]
+        end
+        result = test_multi_objective(multi_objective_1, [0.0, 1.0])
+        @test result ≠ nothing
+        println("Solution for Sphere and Rastrigin: ", result)
+        @test result.u[1][1] ≈ 7.88866e-5 atol=1e-3
+        @test result.u[1][2] ≈ 4.96471e-5 atol=1e-3
+    end
+
+    # Test 2: Rosenbrock and Ackley Functions
+    @testset "Rosenbrock and Ackley Functions" begin
+        function multi_objective_2(x, p=nothing)::Vector{Float64}
+            f1 = (1.0 - x[1])^2 + 100.0 * (x[2] - x[1]^2)^2  # Rosenbrock function
+            f2 = -20.0 * exp(-0.2 * sqrt(0.5 * (x[1]^2 + x[2]^2))) - exp(0.5 * (cos(2π * x[1]) + cos(2π * x[2]))) + exp(1) + 20.0  # Ackley function
+            return [f1, f2]
+        end
+        result = test_multi_objective(multi_objective_2, [1.0, 1.0])
+        @test result ≠ nothing
+        println("Solution for Rosenbrock and Ackley: ", result)
+        @test result.u[10][1] ≈ 1.0 atol=1e-3
+        @test result.u[10][2] ≈ 0.999739 atol=1e-3
```

Comment:
I don't understand how does `u` end up a vector of vectors here?

Reply:
I am not completely sure, but I understand that it is the Pareto front values for the objectives.